control-plane/cmd/webhook-kafka/main.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/eventing/pkg/logconfig"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/conversion"
+	"knative.dev/pkg/webhook/resourcesemantics/defaulting"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/apis/eventing"
+	eventingv1alpha1 "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/eventing/v1alpha1"
+)
+
+var types = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
+
+	eventingv1alpha1.SchemeGroupVersion.WithKind("KafkaSink"): &eventingv1alpha1.KafkaSink{},
```

Comment:
I would not call it a Sink. The concept is a `sink`, just like a source is a source in concept. Call it what it is for Kafka is my recommendation, but I have had a lot of pushback on this idea. Folks just want to add the label to everything....

A kafka source would be `KafkaSubscription`
A kafka sink would be a `KafkaTopic`

A kafka channel would make _n_ `KafkaSubscription` for the controlled `KafkaTopic`

Reply:
I suppose the reason why we have the `KafkaSource` is to keep the naming consistent with the eventing concept event source, similarly to why we should name this `KafkaSink` to align with the eventing concept of event sink